packages/@aws-cdk/aws-s3/lib/bucket.ts

```diff
@@ -514,6 +519,10 @@ export abstract class BucketBase extends cdk.Construct implements IBucket {
    * @returns The `iam.PolicyStatement` object, which can be used to apply e.g. conditions.
    */
   public grantPublicAccess(keyPrefix = '*', ...allowedActions: string[]): iam.PolicyStatement {
+    if (this.publicAccessBlocked) {
+      throw new Error('Cannot grant public access when block public access settings are enabled');
```

Comment:
Change "block public access" to "blockPublicAccess"

Reply:
Done